src/Symfony/Component/Cache/Traits/RedisTrait.php

```diff
@@ -308,7 +313,7 @@ public static function createConnection(string $dsn, array $options = [])
                 $params['parameters']['database'] = $params['dbindex'];
             }
             if (null !== $auth) {
-                $params['parameters']['password'] = $auth;
+                $params['parameters']['password'] = \is_array($auth) ? $auth[1] : $auth;
```

Comment:
this part is strange and I'd be surprised if it worked
can you maybe look at how Predis handles ACL, if it does, and then add tests that cover it?